python/tank/platform/environment.py

```diff
@@ -760,8 +834,21 @@ def update_engine_settings(self, engine_name, new_data, new_location):
         for x in tokens:
             engine_data = engine_data.get(x)
 
-        if new_location:
+        # This is ghetto, but is required given the current design of
```

Comment:
this code looks like it's repeated in multiple places, so we should probably unify it